client/test/unit/components/bhBreadCrumb.spec.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * Created by Dedrick Kitamuka on 23/06/2016.
+ */
+describe('component : bhBreadCrumb', function () {
+    var component, scope, componentController, breadCrumb;
+
+    beforeEach(() => {
+        module('pascalprecht.translate');
+        module('ngStorage');
+        module('angularMoment');
+        module('bhima.services');
+        module('bhima.components');
+    });
+
+    beforeEach(inject(function ($rootScope, _$componentController_){
+        scope = $rootScope.$new();
+        componentController = _$componentController_;
+        breadCrumb = {
+            path : [
+                { label: 'path1', link: '#/path1_link_to_go' },
+                { label: 'path2', link: '#/path2_link_to_go' },
+                { label: 'path3', link: '#/path3_link_to_go', current: true }
+            ],
+
+            button : [
+                { icon: 'glyphicon glyphicon-print', label: 'Print', action: function (){return 'Print';} },
+                { icon: 'glyphicon glyphicon-repeat', label: 'Repeat', action: function () { return 'Repeat';}, color: 'btn-danger' },
+                { icon: 'glyphicon glyphicon-refresh', label: 'Refresh', action: function (){return 'Refresh';}}
+            ],
+
+            label : [
+                { icon: 'glyphicon glyphicon-print', label: 'My Label 1' },
+                { label: 'My label 2' },
+                { label: 'My label 3' }
+            ],
+
+            dropdown : [
+                {
+                    label : 'Dropdown 1',
+                    color : 'btn-primary',
+                    option : [
+                        { label : 'Fc', action : function (){ return 'FC';} },
+                        { label : '$', action : function () {return '$'} }
+                    ]
+                },
+
+                {
+                    label : 'Dropdown 2',
+                    color : 'btn-success',
+                    option : [
+                        { label : 'item1 dd2 with a too long text that you can imagine', action : function (){return 'item1 dd2 long';} },
+                        { label : 'item2 dd2', action : function (){ return 'item1 dd2';}}
+                    ]
+                }
+            ]
+        };
+    }));
+
+    it('receives path data correctly', function (){
```

Comment:
Since all these tests use the same component (withouth changing initialization parameters), you might move the component initialization step into the `beforeEach()` call.

```js
let component;

beforeEach(inject(_$componentController_) => {
  component = _$componentController_('bhBreadcrumb', null, angular.copy(breadCrumb));
}));
```
... then, in each test, you can just write ...
```js
it('receives path data correctly', () => {
  expect(component.path).to.have.length(3);
});
```